subsys/pm/device.c

```diff
@@ -101,6 +101,57 @@ int pm_device_action_run(const struct device *dev,
 	return 0;
 }
 
+int pm_device_power_domain_add(const struct device *dev,
+			       const struct device *domain)
+{
+#ifdef CONFIG_PM_DEVICE_POWER_DOMAIN
+	device_handle_t *rv = (device_handle_t *)domain->handles;
```

Comment:
The explicit `(device_handle_t *)` is what is making you miss the qualifier problems, the compiler isn't complaining when you drop the `const` qualifier. When this is correct you won't need the explicit cast.